packages/form/src/components/SwitchButtonField/index.tsx

```diff
@@ -0,0 +1,46 @@
+import type { FieldPath, FieldValues } from '@ultraviolet/form'
+import { SwitchButton } from '@ultraviolet/ui'
+import type { ComponentProps } from 'react'
+import { useController } from 'react-hook-form'
+import type { BaseFieldProps } from '../../types'
+
+type SwitchButtonFieldProps<
+  TFieldValues extends FieldValues,
+  TFieldName extends FieldPath<TFieldValues>,
+> = BaseFieldProps<TFieldValues, TFieldName> &
+  Pick<
+    ComponentProps<typeof SwitchButton>,
+    'leftButton' | 'rightButton' | 'tooltip' | 'onBlur' | 'onFocus'
+  > & {
+    name: FieldPath<TFieldValues>
+  }
+
+export const SwitchButtonField = <
+  TFieldValues extends FieldValues,
+  TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+>({
+  name,
+  leftButton,
+  rightButton,
+  onBlur,
+  onFocus,
+  tooltip,
+}: SwitchButtonFieldProps<TFieldValues, TFieldName>) => {
```

Comment:
We are trying to switch a bit the strategy for from elements, now we want to pick all the props and exclude the one we don't want (here it will be `name`). Then in the TSX we wan't to spread the props. This way if a new prop is added on the component it will be automatically propagated on froms 😄